codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,268 @@
+import argparse
+import logging
+import uuid
+import subprocess
+import getpass
+from pathlib import Path
+import random
+import re
+
+from .worker_manager import WorkerManager, WorkerJob
+
+logger = logging.getLogger(__name__)
+
+
+class SlurmBatchWorkerManager(WorkerManager):
+    NAME = 'slurm-batch'
+    DESCRIPTION = 'Worker manager for submitting jobs using Slurm Batch'
+
+    SRUN_COMMAND = 'srun'
+    SBATCH_COMMAND = 'sbatch'
+    SBATCH_PREFIX = '#SBATCH'
+    SQUEUE_COMMAND = 'squeue'
+    SBATCH_COMMAND_RETRUN_REGEX = re.compile(r'^Submitted batch job (\d+)$')
+
+    @staticmethod
+    def add_arguments_to_subparser(subparser):
+        subparser.add_argument(
+            '--job-name',
+            type=str,
+            default='codalab-slurm-worker',
+            help='Name for the job definitions that will be generated by this worker manager',
+        )
+        subparser.add_argument(
+            '--nodelist', type=str, default='', help='The worker node to run jobs in'
+        )
+        subparser.add_argument(
+            '--partition', type=str, default='jag-standard', help='Name of batch job queue to use'
+        )
+        subparser.add_argument(
+            '--cpus', type=int, default=1, help='Default number of CPUs for each worker'
+        )
+        subparser.add_argument(
+            '--gpus', type=int, default=1, help='Default number of GPUs for each worker'
+        )
+        subparser.add_argument(
+            '--memory-mb', type=int, default=2048, help='Default memory (in MB) for each worker'
+        )
+        subparser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Print out Slurm batch job definition without submitting to Slurm',
+        )
+        subparser.add_argument(
+            '--user', type=str, default=getpass.getuser(), help='User to run the Batch jobs as'
+        )
+        subparser.add_argument(
+            '--password-file',
+            type=str,
+            required=True,
+            help='Path to the file containing the username and '
+            'password for logging into the CodaLab worker '
+            'each on a separate line. If not specified, the '
+            'the worker will fail to start',
+        )
+        subparser.add_argument(
+            '--work-dir',
+            default='slurm-worker-scratch',
+            help='Directory where to store temporary bundle data, '
+            'including dependencies and the data from run bundles',
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+        self.username = self.args.user
+        # A set of newly submitted job id to keep tracking worker status, as worker might not be created right away.
+        self.submitted_jobs = self.load_worker_jobs()
+
+    def load_worker_jobs(self):
+        """
+        Load worker jobs that are created using SlurmWorkerManager and
```

Comment:
not all worker jobs are guaranteed to be created by the `slurmworkermanager`, right? I think you should also check the job name and make sure it matches the pattern created by `create_random_job_name`?

Reply:
Yes, it's currently coded as matching the prefix `username + job_name`. Or maybe I can make it more explicit?

Reply:
Oh oops I was misled--you're right! The comment below with the example output made me think that you were just counting lines from `squeue` without looking at the name. Maybe update the comment, though

Reply:
Hmm, I was doing a partial match on `username` and `job_name`. Let me see how to make this more clear in comments.